sympy/series/limits.py

```diff
@@ -69,8 +70,17 @@ def heuristics(e, z, z0, dir):
         if r:
             rv = e.func(*r)
             if rv is S.NaN:
-                return
-
+                try:
+                    rat_e = ratsimp(e)
+                    if e != rat_e:
+                        e = rat_e if rat_e != S.NaN else e
+                        rv = limit(e, z, z0, dir)
```

Comment:
If `rat_e == nan`, then `e` would remain unchanged. Can this lead to infinite recursion?